editor/src/components/canvas/controls/selection-area-helpers.ts

```diff
@@ -0,0 +1,138 @@
+  )
+
+  return [
+    ...selectableElements,
+    ...nonSelectableElementsPossiblyUnderMouse,
+    ...mapDropNulls(
+      (path) => MetadataUtils.findElementByElementPath(metadata, path),
+      selectedViews,
+    ),
+  ]
+}
+
+export const getElementsUnderSelectionArea = (
+  metadata: ElementInstanceMetadataMap,
+  selectionAreaCanvasRect: CanvasRectangle,
+): ElementPath[] => {
+  const possibleElementsUnderSelectionArea = mapDropNulls((element) => {
+    if (element.globalFrame == null || !isFiniteRectangle(element.globalFrame)) {
+      return null
+    }
+    const isChildOfSceneRoot = MetadataUtils.isProbablyScene(
+      metadata,
+      EP.nthParentPath(element.elementPath, 3),
+    )
+    if (!(isChildOfSceneRoot || EP.isStoryboardPath(EP.parentPath(element.elementPath)))) {
+      return null
+    }
+
+    return {
+      path: element.elementPath,
+      frame: element.globalFrame,
+      type: isChildOfSceneRoot
+        ? 'scene-child'
+        : MetadataUtils.isProbablyScene(metadata, element.elementPath)
+        ? 'scene'
+        : 'storyboard-child',
+    }
+  }, Object.values(metadata))
+
+  const allElementsUnderSelectionArea = mapDropNulls((element) => {
+    if (!rectanglesOverlap(element.frame, selectionAreaCanvasRect)) {
+      return null
+    }
+    return {
+      ...element,
+      fullyCovered: rectangleContainsRectangle(selectionAreaCanvasRect, element.frame),
+    }
+  }, possibleElementsUnderSelectionArea)
+
+  const thereAreStoryboardChildren = allElementsUnderSelectionArea.some(
+    (other) => other.type === 'storyboard-child',
+  )
+
+  return allElementsUnderSelectionArea
+    .filter((element) => {
+      // if the element is a schene child and there are storyboard children, skip it
+      if (element.type === 'scene-child' && thereAreStoryboardChildren) {
+        return false
+      }
+      // if the element is a scene, and the scene is not fully covered skip the scene
+      if (element.type === 'scene' && !element.fullyCovered) {
+        return false
+      }
+      // if a scene is fully covered, select just the scene and omit its children
+      if (element.type === 'scene-child') {
+        const parentScene = allElementsUnderSelectionArea.find(
+          (other) => other.type === 'scene' && EP.isDescendantOf(element.path, other.path),
+        )
+        if (parentScene != null && parentScene.fullyCovered) {
+          return false
+        }
+      }
+      return true
+    })
+    .map((r) => r.path)
+}
+
+export const elementIsUnderMouse =
+  (mousePointOnCanvas: CanvasPoint | null) =>
+  (element: ElementInstanceMetadata | null): boolean => {
+    return mousePointOnCanvas == null || element == null
+      ? false
+      : element.globalFrame != null &&
+          isFiniteRectangle(element.globalFrame) &&
+          rectContainsPoint(element.globalFrame, mousePointOnCanvas)
+  }
+
+export function getSelectionAreaRenderedRect(
```

Comment:
I'm a bit confused by this function. Is this just applying the `canvasOffset` to the `selectionArea`?

Reply:
kinda, but not just the canvas offset - by using the bounding rect it also takes into account both the navigator _and_ the code editor (but maybe there's a better way?)

Reply:
if makeSelectionArea returns a WindowRectangle, and this here takes a WindowRectangle, and then this thing returns a WindowRectangle, that sounds more correct to me